src/java/org/apache/cassandra/locator/ReplicaPlan.java

```diff
@@ -106,18 +106,25 @@ ForTokenRead withContact(EndpointsForToken newContact)
     public static class ForRangeRead extends ForRead<EndpointsForRange>
     {
         final AbstractBounds<PartitionPosition> range;
+        final int rangeCount;
 
-        public ForRangeRead(Keyspace keyspace, ConsistencyLevel consistencyLevel, AbstractBounds<PartitionPosition> range, EndpointsForRange candidates, EndpointsForRange contact)
+        public ForRangeRead(Keyspace keyspace, ConsistencyLevel consistencyLevel, AbstractBounds<PartitionPosition> range, EndpointsForRange candidates, EndpointsForRange contact, int rangeCount)
         {
             super(keyspace, consistencyLevel, candidates, contact);
             this.range = range;
+            this.rangeCount = rangeCount;
         }
 
         public AbstractBounds<PartitionPosition> range() { return range; }
 
+        /**
+         * @return number of vnode ranges
```

Comment:
Perhaps we could extend this to `number of vnode ranges intersected by the range`, or rename the method to `vnodesCount`/`subrangeCount`, or something like that. I think that having a singular `range` and a range count at the same time might be a bit confusing to unaware readers.

Reply:
good idea. updated the javadoc and method name to `vnodeCount`